src/components/MemberComponent.js

```diff
@@ -0,0 +1,317 @@
+import React, { useReducer, useEffect } from "react";
+import {
+  Card,
+  CardHeader,
+  CardContent,
+  List,
+  Modal,
+  Typography,
+  Button,
+  ListItem,
+  ListItemText,
+  ListItemSecondaryAction,
+  IconButton,
+  Container,
+  TextField,
+} from "@material-ui/core";
+import DeleteIcon from "@material-ui/icons/Delete";
+import AddIcon from "@material-ui/icons/Add";
+import * as db from "../dbUtils";
+
+const MemberComponent = (refreshProjects, project, displayPopup) => {
+  const initialState = {
+    memberList: null,
+    addMode: false,
+    editMode: false,
+    addMember: false,
+    addMemberError: "",
+    isEditing: false,
+    editingIndex: null,
+  };
+
+  const [state, setState] = useReducer(
+    (state, newState) => ({ ...state, ...newState }),
+    initialState
+  );
+
+  const initialNewMember = {
+    email: "",
+    firstName: "",
+    lastName: "",
+  };
+
+  const [newMember, setNewMember] = useReducer(
+    (newMember, newState) => ({ ...newMember, ...newState }),
+    initialNewMember
+  );
+
+  useEffect(() => {
+    if (project) {
+      const { memberList } = project;
+      setState({ memberList });
+    } else displayPopup("Project not loaded");
```

Comment:
This is where you are getting the warning

Reply:
There are 2 ways around this, either leave a comment that you can find in some of mine (//es-lint...) or you load the function in a useEffect which is WAY more complicated than it has to be. Basically the issue is that it says there is a chance that this function might not be loaded before the useEffect fires but a lot of the community suppresses the warnings (we even had it in a JS lab, 15 or 16 I think). Other way is to just smack the contents of the method in there but suppressing the working works fine